crates/oxc_minifier/src/peephole/minimize_statements.rs

```diff
@@ -0,0 +1,380 @@
+                if let Some(Statement::ExpressionStatement(prev_expr_stmt)) = result.last_mut() {
+                    let a = &mut prev_expr_stmt.expression;
+                    let b = &mut throw_stmt.argument;
+                    throw_stmt.argument = Self::join_sequence(a, b, ctx);
+                    result.pop();
+                    self.mark_current_function_as_changed();
+                }
+                result.push(Statement::ThrowStatement(throw_stmt));
+                *is_control_flow_dead = true;
+            }
+            Statement::ForStatement(mut for_stmt) => {
+                match result.last_mut() {
+                    Some(Statement::ExpressionStatement(prev_expr_stmt)) => {
+                        if let Some(init) = &mut for_stmt.init {
+                            if let Some(init) = init.as_expression_mut() {
+                                let a = &mut prev_expr_stmt.expression;
+                                *init = Self::join_sequence(a, init, ctx);
+                                result.pop();
+                                self.mark_current_function_as_changed();
+                            }
+                        } else {
+                            for_stmt.init = Some(ForStatementInit::from(
+                                ctx.ast.move_expression(&mut prev_expr_stmt.expression),
+                            ));
+                            result.pop();
+                            self.mark_current_function_as_changed();
+                        }
+                    }
+                    Some(Statement::VariableDeclaration(prev_var_decl)) => {
+                        if let Some(init) = &mut for_stmt.init {
+                            if prev_var_decl.kind.is_var() {
+                                if let ForStatementInit::VariableDeclaration(var_decl) = init {
+                                    if var_decl.kind.is_var() {
+                                        var_decl
+                                            .declarations
+                                            .splice(0..0, prev_var_decl.declarations.drain(..));
+                                        result.pop();
+                                        self.mark_current_function_as_changed();
+                                    }
+                                }
+                            }
+                        } else if prev_var_decl.kind.is_var() {
+                            let var_decl = ctx.ast.move_variable_declaration(prev_var_decl);
+                            for_stmt.init = Some(ForStatementInit::VariableDeclaration(
+                                ctx.ast.alloc(var_decl),
+                            ));
+                            result.pop();
+                            self.mark_current_function_as_changed();
+                        }
+                    }
+                    _ => {}
+                }
+                result.push(Statement::ForStatement(for_stmt));
+            }
```

Comment:
(I guess you're aware of it, but in case) I think `ForInStatement`, `LabeledStatement`, `BlockStatement` is missing here.
https://github.com/oxc-project/oxc/blob/29417ddc03085a49caaf31b2a6477a24cf4544f5/crates/oxc_minifier/src/peephole/statement_fusion.rs#L70-L80
https://github.com/oxc-project/oxc/blob/29417ddc03085a49caaf31b2a6477a24cf4544f5/crates/oxc_minifier/src/peephole/statement_fusion.rs#L134-L150
Maybe `BlockStatement` should be handled in a different place though.